agenta-backend/agenta_backend/services/event_db_manager.py

```diff
@@ -134,31 +117,314 @@ async def create_trace_span(payload: CreateSpan) -> str:
+    sorters: SorterParams,
+) -> List[Trace]:
+    """Get the traces for the given app_id.
+
+    Args:
+        app_id (str): The ID of the app
+        pagination (PaginationParam): The data of the pagination param
+        filters_param (GenerationFilterParams): The data of the generation filter params
+        sorters (SorterParams): The data of the sorters param
+
+    Returns:
+        List[Trace]: the list of trace for the given app_id
+    """
+
+    # Apply pagination and sorting
+    skip, limit = get_pagination_skip_limit(pagination)
+    sort_direction = (
+        pymongo.ASCENDING if sorters.created_at == "asc" else pymongo.DESCENDING
+    )
+
+    # Fetch traces without pagination and sorting applied
+    base_traces_db = TraceDB.find(
+        TraceDB.app_id == app_id,
+    )
+
+    # Count of traces in db
+    filters_query = {}
+    if filters_param.environment is not None:
+        filters_query["environment"] = filters_param.environment
+    # elif filters_param.variant is not None: TODO: improve in observability sdk
+    #     filters_query["base_id"] = filters_param.variant
+    traces_count = await base_traces_db.find(filters_query, fetch_links=True).count()
+
+    # Fetch traces with pagination and sorting applied
+    traces_db = (
+        await base_traces_db.find(fetch_links=True, skip=skip, limit=limit)
+        .sort([(TraceDB.created_at, sort_direction)])
+        .to_list()
+    )
+
+    # Convert beanie documents to pydantic models and filter based on the filter_params
+    traces = await traces_to_pydantic(traces_db)
+    filtered_traces = filter(
+        partial(filters.filter_document_by_filter_params, filters_param), traces
+    )
+
+    return get_paginated_data(list(filtered_traces), traces_count, pagination)
+
+
+async def fetch_trace_detail(trace_id: str) -> TraceDetail:
+    """Get a trace detail.
+
+    Args:
+        trace_id (str): The ID of a trace
+        user_uid (str): The user ID
+
+    Returns:
+        TraceDetail: trace detail pydantic model
+    """
+
+    trace_db = await get_single_trace(trace_id)
+    app_variant_db = await db_manager.fetch_app_variant_by_id(trace_db.variant_id)
+
+    span_status = (
+        SpanStatus(value=trace_db.status)
+        if trace_db.status in ["INITIATED", "COMPLETED"]
+        else SpanStatus(value=None, error=Error(message=trace_db.status))
+    )
+    return TraceDetail(
+        **{
+            "id": str(trace_db.id),
+            "content": {
+                "inputs": [
+                    {"input_name": key, "input_value": value}
+                    for key, value in trace_db.inputs.items()
+                ],
+                "outputs": trace_db.outputs,
+            },
+            "created_at": trace_db.created_at.isoformat(),
+            "variant": {
+                "variant_id": str(app_variant_db.id),
+                "variant_name": app_variant_db.variant_name,
+                "revision": app_variant_db.revision,
```

Comment:
see detail below. But in any case, I think we should not save any information in the trace anymore (see document). Should be simple to do